src/SBarcodeDecoder.cpp

```diff
@@ -142,9 +144,16 @@ void SBarcodeDecoder::process(const QImage capturedImage, ZXing::BarcodeFormats
     setIsDecoding(false);
 }
 
-QImage SBarcodeDecoder::videoFrameToImage(QVideoFrame &videoFrame, const QRect &captureRect)
+QImage SBarcodeDecoder::videoFrameToImage(const QVideoFrame &videoFrame, const QRect &captureRect)
 {
-    if (videoFrame.handleType() == QAbstractVideoBuffer::NoHandle) {
+
+    #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+
```

Comment:
This define basically splits the whole function in half - I think it would be clearer if we define 2 private functions that have the implementation without ifdefs and one public function to call them  according to Qt version

Reply:
Instead of using two seperate functions which is doing the same thing, it would be good choice to apart them using #ifdef directives in a single function.

Reply:
Well now that the Qt6 part is shrter I guess it's okay to leave it that way. 
But for general case I think it's safer to write a dispather function with `#if` that would call specialized versions without any macros. Espetially for N>2 cases